connector/src/main/java/org/geysermc/connector/utils/DimensionUtils.java

```diff
@@ -0,0 +1,101 @@
+package org.geysermc.connector.utils;
+
+import com.nukkitx.math.vector.Vector3i;
+import com.nukkitx.protocol.bedrock.packet.*;
+import org.geysermc.connector.entity.Entity;
+import org.geysermc.connector.network.session.GeyserSession;
+import org.geysermc.connector.network.translators.TranslatorsInit;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DimensionUtils {
+    public static void switchDimension(GeyserSession session, int javaDimension, boolean fake) {
+        int bedrockDimension = javaToBedrock(javaDimension);
+        Entity player = session.getPlayerEntity();
+        if (bedrockDimension == player.getDimension())
+            return;
+
+        Vector3i pos = Vector3i.from(0, 32767, 0);
+
+        List<Entity> entities = new ArrayList<>(session.getEntityCache().getEntities().values());
+        for (Entity entity : entities) {
+            session.getEntityCache().removeEntity(entity, false);
+        }
+
+        ChangeDimensionPacket changeDimensionPacket = new ChangeDimensionPacket();
+        changeDimensionPacket.setDimension(bedrockDimension);
+        changeDimensionPacket.setRespawn(true);
+        changeDimensionPacket.setPosition(pos.toFloat());
+        session.getUpstream().sendPacket(changeDimensionPacket);
+        player.setDimension(bedrockDimension);
+
+        //let java server handle portal travel sound
+        StopSoundPacket stopSoundPacket = new StopSoundPacket();
+        stopSoundPacket.setStoppingAllSound(true);
+        stopSoundPacket.setSoundName("");
+        session.getUpstream().sendPacket(stopSoundPacket);
+
+        EntityEventPacket eventPacket = new EntityEventPacket();
+        eventPacket.setRuntimeEntityId(player.getGeyserId());
+        eventPacket.setEvent(EntityEventPacket.Event.RESPAWN);
+        eventPacket.setData(0);
+        session.getUpstream().sendPacket(eventPacket);
+
+        session.setSpawned(false);
+        session.setSwitchingDim(true);
+
+        if (fake) {
+            try {
+                Thread.sleep(250);
```

Comment:
The reason I am sleeping the thread instead of using a new thread is because it is important the java packets are not processed before this method returns.

Reply:
What about for other players?

Reply:
I just tested to confirm that each GeyserSession has its own thread for translating java packets.

Reply:
I forgot we did that, my mistake.

Reply:
I can understand that you forgot that. I added that once as a temp. fix and we would change it later on but we never did.